src/libraries/CalldataDecoder.sol

```diff
@@ -178,6 +178,19 @@ library CalldataDecoder {
         }
     }
 
+    /// @dev equivalent to: abi.decode(params, (Currency, address, uint256)) in calldata
+    function decodeCurrencyAddressAndUint256(bytes calldata params)
+        internal
+        pure
+        returns (Currency currency, address _address, uint256 amount)
+    {
+        assembly ("memory-safe") {
```

Comment:
I suggest reverting if the data is < 96 bytes to mimic the behaviour of `abi.encode`

Reply:
we will address this in a separate PR